app/src/sharedTest/java/org/oppia/app/completedstorylist/CompletedStoryListActivityTest.kt

```diff
@@ -0,0 +1,202 @@
+package org.oppia.app.completedstorylist
+
+import android.app.Application
+import android.content.Context
+import android.content.Intent
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.core.app.ActivityScenario.launch
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.contrib.RecyclerViewActions.scrollToPosition
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import org.hamcrest.CoreMatchers.containsString
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.model.ProfileId
+import org.oppia.app.recyclerview.RecyclerViewMatcher.Companion.atPositionOnView
+import org.oppia.domain.topic.StoryProgressTestHelper
+import org.oppia.util.logging.EnableConsoleLog
+import org.oppia.util.logging.EnableFileLog
+import org.oppia.util.logging.GlobalLogLevel
+import org.oppia.util.logging.LogLevel
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+/** Tests for [CompletedStoryListActivity]. */
+@RunWith(AndroidJUnit4::class)
+class CompletedStoryListActivityTest {
+
+  private val internalProfileId = 0
+  @Inject lateinit var storyProfileTestHelper: StoryProgressTestHelper
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    setUpTestApplicationComponent()
+
+    val profileId = ProfileId.newBuilder().setInternalId(internalProfileId).build()
+    storyProfileTestHelper.markFullStoryProgressForFractions(profileId)
+    storyProfileTestHelper.markFullStoryPartialTopicProgressForRatios(profileId)
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerCompletedStoryListActivityTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
+  @Test
+  fun testCompletedStoryList_checkItem0_titleIsCorrect() {
```

Comment:
Here in title you mentioned item 0. But you are testing for item 1. Please check this.

Reply:
Resolved.